packages/Webkul/Product/src/Repositories/ProductRepository.php

```diff
@@ -330,7 +332,7 @@ public function getNewProducts()
                 ->where('product_flat.channel', $channel)
                 ->where('product_flat.locale', $locale)
                 ->inRandomOrder();
-        })->paginate(4);
+        })->paginate($count);
```

Comment:
@deepaksinghgusain227 
In case 0 is set in $count then default reset to 4 again for default theme.